r2/r2/lib/media.py

```diff
@@ -312,6 +312,16 @@ def _scrape_media(url, autoplay=False, maxwidth=600, force=False,
             print "%s made a bad secure media obj for url %s" % (scraper, url)
             secure_media_object = None
 
+        # If thumbnail can't be found, attempt again using _ThumbnailOnlyScraper
+        # This should fix bugs that occur when embed.ly caches links before the thumbnail is available
+        if not thumbnail_image:
+            scraper = _ThumbnailOnlyScraper(url)
+            try:
+                thumbnail_image, preview_object, _, _ = (
```

Comment:
Another minor style nit here: no need to wrap these lines. That was only done elsewhere because the long variable names put it over that line-length limit. So this can just be:

    thumbnail_image, preview_object, _, _ = scraper.scrape()